source/nodejs/adaptivecards/src/carousel.ts

```diff
@@ -417,6 +450,41 @@ export class Carousel extends Container {
             }
         );
     }
+
+    private createCarouselEvent (type : Enums.CarouselInteractionEvent): CarouselEvent
+    {
+        let currentPageId : string | undefined;
+        if (this.currentPageIndex != undefined) {
+            currentPageId = this.getItemAt(this.currentPageIndex).id;
+        }
+        return new CarouselEvent(type, this.id, currentPageId, this.currentPageIndex);   
+    }
+
+    private raiseCarouselEvent(eventType : Enums.CarouselInteractionEvent) {
+        const onCarouselEventHandler = AdaptiveCard.onCarouselEvent;
+        if (onCarouselEventHandler && eventType == Enums.CarouselInteractionEvent.Pagination) {
+            onCarouselEventHandler(this.createCarouselEvent(this.currentEventType));
```

Comment:
Can you have comment in code that why you didn't use eventType directly, instead, you used this currentEventType?

```
 onCarouselEventHandler(this.createCarouselEvent(eventType));
```